src/frequenz/client/dispatch/_client.py

```diff
@@ -0,0 +1,234 @@
+
+"""Dispatch API client for Python."""
+from datetime import datetime, timedelta
+from typing import Any, Awaitable
+
+import grpc
+from frequenz.api.dispatch.v1 import dispatch_pb2_grpc
+
+# pylint: disable=no-name-in-module
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    DispatchDeleteRequest,
+    DispatchFilter,
+    DispatchGetRequest,
+    DispatchList,
+    DispatchListRequest,
+    DispatchUpdateRequest,
+)
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    TimeIntervalFilter as PBTimeIntervalFilter,
+)
+from google.protobuf.timestamp_pb2 import Timestamp
+
+from ._types import (
+    ComponentSelector,
+    Dispatch,
+    DispatchCreateRequest,
+    RecurrenceRule,
+    _component_selector_to_protobuf,
+)
+
+# pylint: enable=no-name-in-module
+
+
+class Client:
+    """Dispatch API client."""
+
+    def __init__(self, grpc_channel: grpc.aio.Channel, svc_addr: str) -> None:
+        """Initialize the client.
+
+        Args:
+            grpc_channel: gRPC channel to use for communication with the API.
+            svc_addr: Address of the service to connect to.
+        """
+        self._svc_addr = svc_addr
+        self._stub = dispatch_pb2_grpc.MicrogridDispatchServiceStub(grpc_channel)
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    async def list(
+        self,
+        microgrid_id: int,
+        component_selectors: list[ComponentSelector] | None = None,
+        start_from: datetime | None = None,
+        start_to: datetime | None = None,
+        end_from: datetime | None = None,
+        end_to: datetime | None = None,
+        is_active: bool | None = None,
+        is_dry_run: bool | None = None,
+    ) -> list[Dispatch]:
+        """List dispatches.
+
+        Args:
+            microgrid_id: The microgrid_id to list dispatches for.
+            component_selectors: optional, list of component ids or categories to filter by.
+            start_from: optional, filter by start_time >= start_from.
+            start_to: optional, filter by start_time < start_to.
+            end_from: optional, filter by end_time >= end_from.
+            end_to: optional, filter by end_time < end_to.
+            is_active: optional, filter by is_active status.
+            is_dry_run: optional, filter by is_dry_run status.
+
+        Returns:
+            DispatchList: List of dispatches.
+        """
+        time_interval = None
+
+        def to_timestamp(dt: datetime | None) -> Timestamp | None:
+            if dt is None:
+                return None
+
+            ts = Timestamp()
+            ts.FromDatetime(dt)
+            return ts
+
+        if start_from or start_to or end_from or end_to:
+            time_interval = PBTimeIntervalFilter(
+                start_from=to_timestamp(start_from),
+                start_to=to_timestamp(start_to),
+                end_from=to_timestamp(end_from),
+                end_to=to_timestamp(end_to),
+            )
```

Comment:
> But it should be also possible to always create the PB object with all Nones and it will have the same effect?

I'd need to research that to be sure, right now, no idea